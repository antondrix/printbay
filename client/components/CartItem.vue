<template>
  <v-flex class="cart-item">
    <v-card class="mb-4">
      <v-container grid-list-lg>
        <v-layout
          row
          wrap
        >
          <v-flex
            xs12
            md4
          >
            <v-img
              :src="item.image"
              height="225px"
            />
          </v-flex>
          <v-flex
            xs12
            md8
          >
            <v-layout
              column
              align-end
            >
              <v-flex>
                <v-layout row>
                  <v-flex
                    sm12
                    md7
                  >
                    <div class="title body-2 grey--text text--darken-1">
                      {{ item.title }} - {{ item.artist }}
                    </div>
                  </v-flex>
                  <v-flex
                    sm6
                    md2
                  >
                    <v-text-field
                      v-model="qty"
                      type="number"
                      label="Qty"
                    />
                  </v-flex>
                  <v-flex
                    sm6
                    md3
                    text-xs-right
                  >
                    <div class="headline">
                      {{ item.price * item.qty | currency }}
                    </div>
                    <div class="body-1">
                      ({{ item.price | currency }})
                    </div>
                  </v-flex>
                </v-layout>
              </v-flex>
              <v-flex justify-end>
                <v-btn
                  color="blue-grey darken-4 white--text"
                  class="elevation-0 mx-0"
                  @click="removeFromCart"
                >
                  Remove
                </v-btn>
              </v-flex>
            </v-layout>
          </v-flex>
        </v-layout>
      </v-container>
    </v-card>
  </v-flex>
</template>
<script>
import { currency } from "@/filters";
export default {
  name: "CartItem",
  filters: { currency },
  props: {
    id: String
  },
  computed: {
    item () {
      return {};
    },
    qty () {
      return 0;
    }
  },
  methods: {
    removeFromCart () {
      //
    }
  }
};
</script>
