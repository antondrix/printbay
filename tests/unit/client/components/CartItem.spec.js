import CartItem from "@/components/CartItem";
import createWrapper from "../wrapper";
import Item from "@/store/models/Item";
import { VBtn, VTextField } from "vuetify/lib";

jest.mock("@/store/models/Item", () => require("../mocks/Item"));

describe("CartItem.vue", () => {
  const id = "1";
  const config = {
    propsData: {
      id
    },
    stubs: {
      VBtn,
      VTextField
    }
  };
  it("should render correctly", () => {
    const wrapper = createWrapper(CartItem, config);
    expect(wrapper).toMatchSnapshot();
  });
  it("should call item update when remove button clicked", () => {
    const spy = jest.spyOn(Item, "update");
    createWrapper(CartItem, config)
      .find(VBtn)
      .trigger("click");
    expect(spy).toHaveBeenCalledWith({
      where: id,
      data: {
        cart: 0
      }
    });
  });
  it("should change quantity when input changes", () => {
    const wrapper = createWrapper(CartItem, config);
    const spy = jest.spyOn(wrapper.vm, "qty", "set");
    const input = wrapper.find(VTextField).find("input");
    const newQty = "2";
    input.setValue(newQty);
    expect(spy).toHaveBeenLastCalledWith(newQty);
  });
});
