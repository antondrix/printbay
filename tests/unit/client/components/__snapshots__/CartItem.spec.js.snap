// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem.vue should render correctly 1`] = `
<v-flex-stub tag="div" class="cart-item">
  <v-card-stub tag="div" class="mb-4">
    <v-container-stub tag="div" grid-list-lg="">
      <v-layout-stub tag="div" row="" wrap="">
        <v-flex-stub tag="div" xs12="" md4="">
          <v-img-stub height="225px" src="https://images.nga.gov/?service=asset&amp;action=show_preview&amp;asset=149207" position="center center" transition="fade-transition"></v-img-stub>
        </v-flex-stub>
        <v-flex-stub tag="div" xs12="" md8="">
          <v-layout-stub tag="div" column="" align-end="">
            <v-flex-stub tag="div">
              <v-layout-stub tag="div" row="">
                <v-flex-stub tag="div" sm12="" md7="">
                  <div class="title body-2 grey--text text--darken-1">
                    Self-Portrait - Vincent van Gogh
                  </div>
                </v-flex-stub>
                <v-flex-stub tag="div" sm6="" md2="">
                  <div class="v-input v-text-field v-input--is-label-active v-input--is-dirty theme--light">
                    <div class="v-input__control">
                      <div class="v-input__slot">
                        <div class="v-text-field__slot"><label aria-hidden="true" class="v-label v-label--active theme--light" style="left: 0px; position: absolute;">Qty</label><input aria-label="Qty" type="number"></div>
                      </div>
                      <div class="v-text-field__details">
                        <div class="v-messages theme--light">
                          <div class="v-messages__wrapper"></div>
                        </div>
                      </div>
                    </div>
                  </div>
                </v-flex-stub>
                <v-flex-stub tag="div" sm6="" md3="" text-xs-right="">
                  <div class="headline">
                    $0.00
                  </div>
                  <div class="body-1">
                    ($29.95)
                  </div>
                </v-flex-stub>
              </v-layout-stub>
            </v-flex-stub>
            <v-flex-stub tag="div" justify-end=""><button type="button" class="elevation-0 mx-0 v-btn theme--light blue-grey darken-4 white--text">
                <div class="v-btn__content">
                  Remove
                </div>
              </button></v-flex-stub>
          </v-layout-stub>
        </v-flex-stub>
      </v-layout-stub>
    </v-container-stub>
  </v-card-stub>
</v-flex-stub>
`;
